terraform/modules/tf_threat_intel_downloader/iam.tf

```diff
@@ -0,0 +1,144 @@
+// IAM Role: Execution Role
+resource "aws_iam_role" "threat_intel_downloader" {
+  name               = "${var.prefix}_threat_intel_downloader"
+  assume_role_policy = "${data.aws_iam_policy_document.lambda_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: Generic Lambda AssumeRole
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+// IAM Role Policy: Allow lambda function to invoke the Lambda Function
+resource "aws_iam_role_policy" "threat_intel_downloader" {
+  name   = "invoke_lambda_function_role_policy"
+  role   = "${aws_iam_role.threat_intel_downloader.id}"
+  policy = "${data.aws_iam_policy_document.invoke_lambda_function.json}"
+}
+
+// IAM Policy Doc: Allow the lambda function to invoke the Lambda function
+data "aws_iam_policy_document" "invoke_lambda_function" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "lambda:InvokeFunction",
+    ]
+
+    resources = [
+      "arn:aws:lambda:${var.region}:${var.account_id}:function:${var.prefix}_streamalert_threat_intel_downloader",
```

Comment:
I would feed this in via variables vs interpolation

Reply:
you can use outputs from the main `tf_stream_alert` module

Reply:
I am following the same style to interpolate resources across streamalert terraform modules, example in [streamalert iam tf file](https://github.com/airbnb/streamalert/blob/master/terraform/modules/tf_stream_alert/iam.tf#L48)